### DISCUSSION TITLE
Is there a way to restart only the kube-apiserver ?

### DISCUSSION THREAD

**Discussion by sylvainfaivre** [k3s-io/k3s#8353] [Q&A] [answered]
We have a small cluster setup with a small team, and we would like to use token auth file authentication.

I successfully set it up with k3s, but now the problem is kube-apiserver reads the credentials file only on startup, so I have to restart k3s as a whole each time the credentials are modified.

So I'm wondering, is there a way to restart only the kube-apiserver ?

**Comment by brandond** [comment, marked as answer]
No. K3s runs all the Kubernetes and etcd components in a single process; that is part of what allows it to run with less overhead than most other distros. Because they all run in a single process, there is no way to restart only a single component.

**Reply by sylvainfaivre** [reply]
Thanks, and sorry for not reading the docs better before asking such a basic question.
